pkg/operator/operator.go

```diff
@@ -85,15 +93,43 @@ func (c serviceCAOperator) Key() (metav1.Object, error) {
 
 func (c serviceCAOperator) Sync(obj metav1.Object) error {
 	operatorConfig := obj.(*operatorv1.ServiceCA)
+	setOperatorVersion := false
 
-	switch operatorConfig.Spec.ManagementState {
+	operatorConfigCopy := operatorConfig.DeepCopy()
+	switch operatorConfigCopy.Spec.ManagementState {
 	case operatorv1.Unmanaged, operatorv1.Removed, "Paused":
 		// Totally disable the sync loop in these states until we bump deps and replace sscs.
 		return nil
+	case operatorv1.Managed:
+		// This is to push out deployments but does not handle deployment generation like it used to. It may need tweaking.
+		err := sync_v4_00_to_latest(c, operatorConfigCopy)
+		if err != nil {
+			c.setFailingStatus(operatorConfigCopy, "OperatorSyncLoopError", err.Error())
+		} else {
+			glog.Infof("Sync deployments complete, syncing service-ca-operator status...")
+			setOperatorVersion, err = c.syncStatus(operatorConfigCopy, deploymentNames)
+			if err != nil {
+				return err
+			}
+		}
+		if setOperatorVersion {
+			version := os.Getenv("OPERATOR_IMAGE_VERSION")
+			if c.versionGetter.GetVersions()["operator"] != version {
+				glog.Infof("Updating clusteroperator %s version: %v", clusterOperatorName, version)
+				// Set current version
+				c.versionGetter.SetVersion("operator", version)
+			}
+		}
+	default:
+		glog.Warningf("unknown custom resource state: %s", operatorConfig.Spec.ManagementState)
```

Comment:
I think we shouldn't even try to check the statuses in this case.

Reply:
agreed, thanks, done